packages/medusa/src/services/shipping-profile.ts

```diff
@@ -352,43 +354,75 @@ class ShippingProfileService extends TransactionBaseService {
   }
 
   /**
-   * Adds a product to a profile. The method is idempotent, so multiple calls
-   * with the same product variant will have the same result.
-   * @param profileId - the profile to add the product to.
-   * @param productId - the product to add.
+   * Adds a product of an array of products to the profile.
+   * @param profileId - the profile to add the products to.
+   * @param productId - the ID of the product or multiple products to add.
    * @return the result of update
    */
   async addProduct(
     profileId: string,
-    productId: string
+    productId: string | string[]
   ): Promise<ShippingProfile> {
     return await this.atomicPhase_(async (manager) => {
-      await this.productService_
-        .withTransaction(manager)
-        .update(productId, { profile_id: profileId })
+      const productServiceTx = this.productService_.withTransaction(manager)
+
+      if (Array.isArray(productId)) {
+        for (const pId of productId) {
+          await productServiceTx.update(pId, {
+            profile_id: profileId,
+          })
```

Comment:
thought(non-blocking): Would it make sense to create a dedicated method to handle bulk updating in the service or repo layer? I imagine this operation can get nasty when adding > 10000 products, which is likely to happen when creating new profiles.

Reply:
I think that would make sense, as the current way does not scale. I will add a method to the ProductService that can run the update in a single query instead. 😄

Reply:
I would suggest to do like we do when we migrate the other methods to accept a collection. Like in the cart, line items etc. The update method could accept an id or a collection of ids so it is aligned with the rest of the codebase. Otherwise in some places we will have two different methods and other places one method. A method that handle a collection can handle a collection of one item as well.

Reply:
@adrien2p I think we would need to change up the update product method a bit in order for it to work with more than one product at a time, with regards to how it can be used to update nested relations such as variants. 

While it wouldn't be a breaking change as we can keep the endpoint as is for now, we would open up for people to use the update method in a way that would break. We would need to make the update method accept a map of product ids and updates in order for it to work with variants, to make sure the nested updates only apply to the correct product, which would definitely be a breaking change IMO. So I think it's out of scope for this PR.